draft-ietf-quic-http.md

```diff
@@ -321,8 +321,19 @@ defined in this document: control streams ({{control-streams}}) and push streams
 ({{push-streams}}).  Other stream types can be defined by extensions to HTTP/3;
 see {{extensions}} for more details.
 
-Both clients and servers SHOULD send a value of three or greater for the QUIC
-transport parameter `initial_max_uni_streams`.
+The performance of HTTP/3 connections in the early phase of their lifetime is
+sensitive to the creation and exchange of data on unidirectional streams.
+Endpoints that set low values for the QUIC transport parameters
+`initial_max_uni_streams` and `initial_max_stream_data_uni` will increase the
+chance that the remote peer reaches the limit early. In particular, the value
+chosen for `initial_max_uni_streams` should consider that remote peers may wish
+to exercise reserved stream behaviour ({{stream-grease}}). Limits can be
+increased post handshake at the cost of additional RTT, for example by
+exchanging `STREAMS_BLOCKED` and `MAX_STREAMS` QUIC frames. To reduce the
```

Comment:
STREAMS_BLOCKED has no actual effect, it's just informative, so I don't think it's wise to mention it here.

Reply:
That's a fair point. However, practically speaking, if a peer spends credit legitimately and needs more, how does it get it?

We're seeing various HTTP/3 issues around getting blocked by conservative initial values, and there is little guidance on ways to mitigate this happening. If the answer is to aggressively dole out credit, it kind of devalues having limits.

Reply:
The receiver should know when all the flow control credit is used(ie: it knows the current max and it knows the amount used in order to enforce limits), so when it gets to or close to 0, it should know the peer is unable to send.

In practice, the current text advises providing enough window for 2RTTs I believe, which is reasonably generous.

Reply:
Sorry I should have been more specific, I was speaking about the number of uni streams not data. If the server gives the client initial uni streams value of 1, and the client uses up, what should the server do in the sense of upping the unidirectional stream limit? Not giving sufficient unidirectional streams will cause all sorts of annoying issues at the start of a H3 connection.

> The receiver should know when all the flow control credit is used(ie: it knows the current max and it knows the amount used in order to enforce limits), so when it gets to or close to 0, it should know the peer is unable to send.

> In practice, the current text advises providing enough window for 2RTTs I believe, which is reasonably generous.

The problem is that the flow control credit starts at 0 if not specified. So, without some sensible guidance for an initial value, if a remote peer initiates a new unidirectional stream, it then needs to wait for the the local peer to send a MAX_DATA of _some_ value. Again this is pretty naff when all you are trying to do is create a HTTP/3 Control stream.

Reply:
There's nothing you can do to force a peer to allow more streams. If a peer continues to misbehave, all you can do is close the connection.
STREAMS_BLOCKED is just a debugging frame, so I agree with @ianswett that we should remove it from this PR.